qiskit/visualization/circuit/_utils.py

```diff
@@ -499,42 +499,72 @@ def _sorted_nodes(dag_layer):
     return nodes
 
 
-def _get_gate_span(qubits, node, measure_arrows):
+def _get_gate_span(qubits, clbits, node, measure_arrows):
     """Get the list of qubits drawing this gate would cover
     qiskit-terra #2802
     """
-    min_index = len(qubits)
-    max_index = 0
+    wire_indices = []
     for qreg in node.qargs:
-        index = qubits.index(qreg)
+        try:
+            wire_indices.append(qubits.index(qreg))
+        except ValueError:
+            continue
+
+    offset = len(qubits)
+    for creg in node.cargs:
+        try:
+            wire_indices.append(offset + clbits.index(creg))
+        except ValueError:
+            continue
 
-        if index < min_index:
-            min_index = index
-        if index > max_index:
-            max_index = index
+    condition = getattr(node.op, "condition", None)
+    if condition is not None:
+        for bit in condition_resources(condition).clbits:
+            try:
+                wire_indices.append(offset + clbits.index(bit))
+            except ValueError:
+                continue
+
+    if not wire_indices:
+        # Operations with no wires shouldn't collide with anything.
+        return []
+
+    min_index = min(wire_indices)
+    max_index = max(wire_indices)
 
     if isinstance(node.op, ControlFlowOp) and not isinstance(node.op, BoxOp):
         # Because of wrapping boxes for mpl control flow ops, this
         # type of op must be the only op in the layer
         # BoxOps are excepted because they have one block executed unconditionally
         span = qubits
-    elif node.cargs and (
-        (measure_arrows and isinstance(node.op, Measure)) or getattr(node.op, "condition", None)
-    ):
-        span = qubits[min_index : len(qubits)]
     else:
-        span = qubits[min_index : max_index + 1]
+        wires = list(qubits) + list(clbits)
+        if isinstance(node.op, Measure) and node.cargs and not measure_arrows:
+            # When measure arrows are disabled, the text drawer renders the measurement label on
+            # the qubit wire without spanning the intermediate wires down to the classical target.
+            # Returning just the directly-touched wires prevents stacked measurements on distinct
+            # classical bits from being forced into separate columns when the registers are bundled.
+            span = [wires[index] for index in sorted(set(wire_indices))]
+        else:
+            if (
+                node.cargs
+                and measure_arrows
+                and isinstance(node.op, Measure)
+                and max_index < offset
+            ):
+                max_index = offset - 1
+            span = wires[min_index : max_index + 1]
```

Comment:
This is more of a nitpick: the computation seemed a bit more efficient as it was done previously, in particular `min_index` and `max_index` were computed directly, and there was no need to sort sets. Is it possible to follow a similar strategy when working with `clbits` here as well?